When you just want to do a try-except without handling the exception, how do you do it in Python?Is the following the right way to do it?orThe difference is, that the first one will also catch KeyboardInterrupt, SystemExit and stuff like that, which are derived directly from exceptions.BaseException, not exceptions.Exception. See documentation for details:When you just want to do a try catch without handling the exception, how do you do it in Python?It depends on what you mean by "handling."If you mean to catch it without taking any action, the code you posted will work.If you mean that you want to take action on an exception without stopping the exception from going up the stack, then you want something like this:It\'s generally considered best-practice to only catch the errors you are interested in, in the case of shutil.rmtree it\'s probably OSError:If you want to silently ignore that error, you would do..Why? Say you (somehow) accidently pass the function an integer instead of a string, like..It will give the error "TypeError: coercing to Unicode: need string or buffer, int found" - you probably don\'t want to ignore that, which can be difficult to debug..If you definitely want to ignore all errors, catch Exception rather than a bare except: statement. Again, why?It catches every exception, include the SystemExit exception which sys.exit() uses, for example:..compared to the following, which correctly exits:If you want to write ever better behaved code, the OSError exception can represent various errors, but in the example above we only want to ignore Errno 2, so we could be even more specific:You could also import errno and change the if to if e.errno == errno.ENOENT:For completeness:...from the python tutorial.Also note that you can capture the exception like this:First I quote the answer of Jack o\'Connor from this thread. The referenced thread got closed so I write here:"There\'s a new way to do this coming in Python 3.4:Here\'s the commit that added it: http://hg.python.org/cpython/rev/406b47c64480And here\'s the author, Raymond Hettinger, talking about this and all sorts of other Python hotness (relevant bit at 43:30): http://www.youtube.com/watch?v=OSGv2VnC0go "My addition to this is the Python 2.7 equivalent:Then you use it like in Python 3.4:There are several ways of doing this. However, the choice of example has a simple solution that does not cover the general case.Instead of Do this:This is an argument specific to shutil.rmtree. You can see the help on it by doing the following, and you\'ll see it can also allow for functionality on errors as well.Since this only covers the narrow case of the example, I\'ll further demonstrate how to handle this if those keyword arguments didn\'t exist.Since the above only covers the narrow case of the example, I\'ll further demonstrate how to handle this if those keyword arguments didn\'t exist.You can import the suppress context manager:But only suppress the most specific exception:You will silently ignore a FileNotFoundError:From the docs:As with any other mechanism that completely suppresses exceptions,\n  this context manager should be used only to cover very specific errors\n  where silently continuing with program execution is known to be the\n  right thing to do.Note that suppress and FileNotFoundError are only available in Python 3.If you want your code to work in Python 2 as well, see the next section:When you just want to do a try/except without handling the exception,\n  how do you do it in Python?Is the following the right way to do it?For Python 2 compatible code, pass is the correct way to have a statement that\'s a no-op. But when you do a bare except:, that\'s the same as doing except BaseException: which includes GeneratorExit, KeyboardInterrupt, and SystemExit, and in general, you don\'t want to catch those things. In fact, you should be as specific in naming the exception as you can. Here\'s part of the Python (2) exception hierarchy, and as you can see, if you catch more general Exceptions, you can hide problems you did not expect:You probably want to catch an OSError here, and maybe the exception you don\'t care about is if there is no directory. We can get that specific error number from the errno library, and reraise if we don\'t have that:Note, a bare raise raises the original exception, which is probably what you want in this case. Written more concisely, as we don\'t really need to explicitly pass with code in the exception handling:@When you just want to do a try catch without handling the exception, how do you do it in Python?This will help you to print what exception is:( i.e. try catch without handling the exception and print the exception.)reg,\nTilokchanFYI the else clause can go after all exceptions and will only be run if the code in the try doesn\'t cause an exception. in python, we handle exceptions similar to other language but the difference is some syntex difference, for example-Handling an exception in python: If you have some suspicious code that may raise an exception, you can defend your program by placing the suspicious code in a try: block. 